presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -291,10 +291,19 @@ void PrestoExchangeSource::processDataResponse(
     }
   }
 
-  int64_t ackSequence =
-      atol(headers->getHeaders()
-               .getSingleOrEmpty(protocol::PRESTO_PAGE_NEXT_TOKEN_HEADER)
-               .c_str());
+  std::optional<int64_t> ackSequenceOpt;
```

Comment:
If we are not supporting data fetching from java, then we can get away by not using std::optional. The token check is performed anyways in the else block so whenever receiving data from coordinator it will not continue execution.